e2e/volume_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 OnMetal authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+
+	corev1alpha1 "github.com/onmetal/onmetal-api/api/core/v1alpha1"
+	storagev1alpha1 "github.com/onmetal/onmetal-api/api/storage/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	. "github.com/onsi/gomega"
+
+
+)
+
+var _ = Describe("cephlet-volume", func() {
+
+
+	/*var (
+	 	volumeClass *storagev1alpha1.VolumeClass
+	 	//volumePool  *storagev1alpha1.VolumePool
+		//rookConfig                 *rook.Config
+		//volumePoolSecretAnnotation = "ceph-client-secret-name"
+	 )*/
+
+	const (
+		volumeSize = "10Gi"
+	//	cephClientSecretValue = "test"
+	//	snapshotSize  = "2Gi"
+	//	cephPoolName  = "ceph"
+	//	cephImageName = "image-1"
+	)
+
+
+	It("should create volume", func(ctx SpecContext) {
+		By("checking that a Volume has been created")
+		vol := &storagev1alpha1.Volume{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "tsi",
+				Namespace: "rook-ceph",
+			},
+			Spec: storagev1alpha1.VolumeSpec{
+				VolumeClassRef: &corev1.LocalObjectReference{Name: "fast"},
+				VolumePoolRef:  &corev1.LocalObjectReference{Name: "ceph"},
+				Resources: corev1alpha1.ResourceList{
+					corev1alpha1.ResourceStorage: resource.MustParse(volumeSize),
+				},
+			Image: "ghcr.io/onmetal/onmetal-image/gardenlinux:rootfs-dev-20230223",
+			},
+		}
+		Expect(k8sClient.Create(ctx, vol)).To(Succeed())
+		fmt.Println("Here the Volume is getting created############", vol.Name)
+	})
```

Comment:
The `onmetal-api` is already tested in a different place. Rather create a `Volume` and check in the ceph cluster for instance if a `ceph image` got created, the image has the correct meta data and so on.

Reply:
Okay.. We will add that part as well.